### PR TITLE
Cleaned up javadoc...

### PR DESCRIPTION
**Pull request by yfarjoun**
...so that there are no warnings and so that the links work properly.

- removed unneeded imports
- some IDE-recommended changes.

### Description

_Give your PR a **concise** yet **descriptive** title_
_Please explain the changes you made here._
_Explain the **motivation** for making this change. What existing problem does the pull request solve?_
_Mention any issues fixed, addressed or otherwise related to this pull request, including issue numbers or hard links for issues in other repos._
_You can delete these instructions once you have written your PR description._

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/main/java/picard/util/SingleBarcodeDistanceMetric.java [hunk 1]
```diff
@@ -86,7 +86,8 @@ public int hammingDistance() {
 
     /**
      * Similar to Hamming distance but this version doesn't penalize matching bases with low quality for the read.
-     * @return
+
+     * @return the edit distance between the barcode(s) and the read(s)
      */
     public int leniantHammingDistance() {
```
Comment (pshapiro4broad) [inline src/main/java/picard/util/SingleBarcodeDistanceMetric.java]: typo `lenient` :(

File: src/main/java/picard/sam/SamErrorMetric/BaseCalculator.java [hunk 2]
```diff
@@ -28,8 +28,8 @@
 import htsjdk.samtools.util.SamLocusIterator;
 
 /**
- * An interface that can take a collection of bases (provided as {@link htsjdk.samtools.util.SamLocusIterator.RecordAndOffset RecordAndOffset}
- * and {@link SamLocusAndReferenceIterator.SAMLocusAndReference SAMLocusAndReference}) and generates a
+ * An interface that can take a collection of bases (provided as {@link htsjdk.samtools.util.SamLocusIterator.RecordAndOffset}
+ * and {@link htsjdk.samtools.reference.SamLocusAndReferenceIterator.SAMLocusAndReference}) and generates a
```
Comment (pshapiro4broad) [inline src/main/java/picard/sam/SamErrorMetric/BaseCalculator.java]: The fully qualified name shouldn't be required here since the file imports `htsjdk.samtools.reference.SamLocusAndReferenceIterator`

File: src/main/java/picard/analysis/CollectWgsMetrics.java [hunk 3]
```diff
@@ -57,6 +57,7 @@
 import picard.filter.CountingFilter;
 import picard.filter.CountingMapQFilter;
 import picard.filter.CountingPairedFilter;
+import picard.sam.markduplicates.CollectDuplicateMetrics;
```
Comment (pshapiro4broad) [inline src/main/java/picard/analysis/CollectWgsMetrics.java]: Why is this needed?

**Comment by yfarjoun** [comment]
this has been awaiting your re-review for a while, @pshapiro4broad
